src/test/java/org/springframework/data/release/infra/LicenseHeaderCommandsUnitTests.java

```diff
@@ -0,0 +1,277 @@
+		}
+
+		@Test
+		void headerWithStartAndEndDateUpdatedToPresent() {
+			var originalContent = openSourceJavaFile("2022-2024");
+			var updatedContent = commands.updateLicenseHeaderInFile(mockJavaFile, originalContent, "present",
+					SupportStatus.COMMERCIAL);
+			var expectedContent = commercialJavaFile("2022");
+			assertThat(updatedContent).isEqualTo(expectedContent);
+		}
+
+		@Test
+		void headerUpdateIsIdempotent() {
+			var originalContent = openSourceJavaFile("2022-2024");
+			var updatedContent = commands.updateLicenseHeaderInFile(mockJavaFile, originalContent, "present",
+					SupportStatus.COMMERCIAL);
+			updatedContent = commands.updateLicenseHeaderInFile(mockJavaFile, updatedContent, "present",
+					SupportStatus.COMMERCIAL);
+			var expectedContent = commercialJavaFile("2022");
+			assertThat(updatedContent).isEqualTo(expectedContent);
+		}
+	}
+
+	private String openSourceJavaFile(String dateRange) {
+		return """
+				/*
+				 * Copyright %s the original author or authors.
+				 *
+				 * Licensed under the Apache License, Version 2.0 (the "License");
+				 * you may not use this file except in compliance with the License.
+				 * You may obtain a copy of the License at
+				 *
+				 *      https://www.apache.org/licenses/LICENSE-2.0
+				 *
+				 * Unless required by applicable law or agreed to in writing, software
+				 * distributed under the License is distributed on an "AS IS" BASIS,
+				 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+				 * See the License for the specific language governing permissions and
+				 * limitations under the License.
+				 */
+				package some.package;
+				
+				import some.otherpackage.Foo;
+				
+				/**
+				 * This is some class comment.
+				 */
+				public class SomeClass {
+				}""".formatted(dateRange);
+	}
+
+	private String commercialJavaFile(String startDate) {
+		return """
+				/*
+				 * Copyright %s Broadcom Inc. and/or its subsidiaries. All Rights Reserved.
+				 * Copyright %s-present the original author or authors.
+				 *
+				 *
+				 *
+				 *
+				 *
+				 *
+				 *
+				 *
+				 *
+				 *
+				 *
+				 */
+				package some.package;
+				
+				import some.otherpackage.Foo;
+				
+				/**
+				 * This is some class comment.
+				 */
+				public class SomeClass {
+				}""".formatted(startDate, startDate);
+	}
+
+	private String openSourcePomFile(String dateRange) {
+		return """
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <name>Spring Data Release Train - BOM Infrastructure</name>
+    <licenses>
+        <license>
+            <name>Apache License, Version 2.0</name>
+            <url>https://www.apache.org/licenses/LICENSE-2.0</url>
+            <comments>
+                Copyright %s the original author or authors.
```

Comment:
Waiting to hear back from internal email thread about what this is supposed to actually look like after. I know this is not 100% complete until then.